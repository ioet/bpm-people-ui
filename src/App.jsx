/* eslint-disable react/prop-types,react/jsx-tag-spacing */
import React from 'react';
import Paper from '@material-ui/core/Paper/Paper';
import withStyles from '@material-ui/core/styles/withStyles';
import Divider from '@material-ui/core/Divider/Divider';
import ListContainer from './component/container/ListContainer';
import CreateUserContainer from './component/container/CreateUserContainer';
import ErrorSnackbarContainer from './component/container/ErrorSnackbarContainer';
import Footer from './component/presentational/Footer';

const styles = theme => ({
  root: {
    marginTop: theme.spacing.unit * 3,
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 1,
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: theme.spacing.unit * 2,
    width: '50%',
    margin: 'auto',
  },
});

function App(props) {
  const { classes } = props;
  return (
    <div>
      <Paper className={classes.root} elevation={1}>
        <CreateUserContainer/>
        <Divider/>
        <ListContainer/>
        <Footer />
      </Paper>
      <ErrorSnackbarContainer/>
    </div>
  );
}

export default withStyles(styles)(App);
