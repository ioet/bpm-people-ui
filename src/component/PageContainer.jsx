import React from 'react';
import MenuPanel from './MenuPanel';
import styles from '../generalStyle.css';
import BodyPanel from './BodyPanel';

class PageContainer extends React.Component {
  constructor() {
    super();

    this.state = {
      bodyTitle: 'Nothing',
    };
  }


  render() {
    return (
      <div className={styles.pageContainerStyle}>
        <MenuPanel />
        <BodyPanel />
      </div>

    );
  }
}

export default PageContainer;
