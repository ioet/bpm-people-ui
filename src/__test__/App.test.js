import React from 'react';
import App from '../App';
import PageContainer from '../component/PageContainer';

describe('<App/>', () => {
  test('Component Renders correctly', () => {
    const appWrapper = global.shallow(<App />);

    expect(global.toJson(appWrapper)).toMatchSnapshot();

    expect(appWrapper.find(PageContainer).exists()).toBe(true);
  });
});
