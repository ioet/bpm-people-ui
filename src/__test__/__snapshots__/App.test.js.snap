// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Component Renders correctly 1`] = `
<div>
  <WithStyles(AppBar)
    position="sticky"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        color="inherit"
        variant="h6"
      >
        Here are all users currently saved in our system
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <Connect(WithStyles(UserList)) />
  <WithStyles(Footer) />
  <Connect(WithStyles(BpmButton)) />
  <Connect(WithStyles(ErrorSnackbar)) />
  <Connect(DeleteDialog) />
  <Connect(BpmDialog) />
  <Connect(BpmDialog) />
</div>
`;
